commit a57ecc6e39a49f9010baa58608d6d42f096df932
Author: root <root@DESKTOP-EJUEM8J.localdomain>
Date:   Sun Apr 15 19:01:13 2018 -0400

    Adding new folder

diff --git a/swagger-docker/Dockerfile b/swagger-docker/Dockerfile
new file mode 100644
index 0000000..915cc04
--- /dev/null
+++ b/swagger-docker/Dockerfile
@@ -0,0 +1,32 @@
+FROM mongo:latest
+EXPOSE 27017
+FROM ubuntu:16.04
+
+RUN apt-get update -q && apt-get install -qy \
+    build-essential \
+    python-dev \
+    python-pip \
+    wget \
+    curl \
+    git-core \
+    default-jre \
+    software-properties-common -y
+
+#RUN apt-get update -q && apt-get install -qy default-jre 
+
+
+
+RUN pip install connexion
+RUN pip install pymongo
+RUN pip install six
+RUN pip install json
+RUN git clone https://github.com/cloudmesh-community/hid-sp18-418.git
+
+WORKDIR /hid-sp18-418/swagger-docker
+
+RUN wget http://central.maven.org/maven2/io/swagger/swagger-codegen-cli/2.3.1/swagger-codegen-cli-2.3.1.jar
+EXPOSE 8080
+
+RUN make service
+
+CMD ["make", "start"]
\ No newline at end of file
diff --git a/swagger-docker/Makefile b/swagger-docker/Makefile
new file mode 100644
index 0000000..776fc90
--- /dev/null
+++ b/swagger-docker/Makefile
@@ -0,0 +1,76 @@
+SWAGGERCODEGEN=java -jar swagger-codegen-cli-2.3.1.jar
+YAMLFILE=db.yaml
+DESTDIR=server/db/flaskConnexion
+CREATED=server
+
+all: service start 
+	@echo "DONE"
+
+docker-all: docker-build docker-start
+	@echo "DONE"
+
+docker-build:
+	@echo "building the image from docker file..."
+	docker build -t surya/cloudmesh-dbinfo .
+	@echo "image DONE"
+
+docker-pull:
+	@echo "pulling the image from dockerhub..."
+	docker pull surya/cloudmesh-dbinfo
+	@echo "image DONE"
+
+docker-start:
+	@echo "starting the service in container..."
+	docker run -p 8080:8080 surya/cloudmesh-dbinfo
+
+docker-stop:
+	@echo "stoping the service..."
+	docker stop $$(docker ps -alq)
+	@echo "service stopped"
+
+docker-remove:
+	@echo "removing the image..."
+	docker rmi -f surya/cloudmesh-dbinfo
+	@echo "image removed"
+
+docker-clean: docker-stop docker-remove
+	@echo "DONE"
+
+dest:
+	mkdir -p $(DESTDIR)
+	 
+service: dest
+	@echo "creating the service..."
+	wget http://central.maven.org/maven2/io/swagger/swagger-codegen-cli/2.3.1/swagger-codegen-cli-2.3.1.jar
+	$(SWAGGERCODEGEN) generate \
+	-i $(YAMLFILE) \
+	-l python-flask \
+	-o $(DESTDIR) \
+	-D supportPython3=true
+	
+	rm *.jar
+	pip install --upgrade pip
+
+	cp -r *.py $(DESTDIR)/swagger_server/controllers
+	cp database.py $(DESTDIR)/swagger_server/models
+	cp base_model_.py $(DESTDIR)/swagger_server/models
+	cd $(DESTDIR) && pip install -r requirements.txt && \
+	python setup.py install 
+	
+	
+
+start:  
+	@echo "starting the service..."
+	cd $(DESTDIR) && python -m swagger_server
+
+test:
+	curl http://localhost:8080/api/db
+            
+stop:
+	@echo "stopping the service"
+	ps -ef | grep swagger_server | grep -v "grep" | awk '{print $$2}' | xargs kill -9
+	@echo "service stopped"
+
+clean:
+	@echo "removing service files created"
+	rm -rf $(CREATED)
